eventsourcing/src/test/java/org/axonframework/eventsourcing/AsyncEventSourcingRepositoryTest.java

```diff
@@ -61,7 +61,7 @@ void setUp() {
         testSubject = new AsyncEventSourcingRepository<>(
                 eventStore,
                 identifier -> TEST_MODEL_CRITERIA,
-                (currentState, event) -> currentState + "-" + event.getPayload(),
+                (currentState, event, __) -> currentState + "-" + event.getPayload(),
```

Comment:
I understand the tendency from kotlin to name unused parameters `_`. However, I've never seen it in Java. I think we should just name this `context` or `ctxt`, as it's still a variable in Java.

This, for example, would work:
```+ __.getResource("myKey")```

Reply:
It's a new Java feature in some of the recent version of Java. In the previous versions you weren't able to do that (eg. Java 8), so I think it's the case you've never seen that. 
So I think it's perfectly fine if you don't need the variable - you won't see the unused argument blanked in your IDE (IntelliJ supports this notation).

Reply:
Sorry, I mixed two things, since Java 22 you are able to do `_` just underscore, but we'are here on Java 21, so `__` it's a workaround while waiting for the language support.

Reply:
Interesting, it's a preview feature in 21: https://docs.oracle.com/en/java/javase/21/language/unnamed-variables-and-patterns.html

It was finalized in 22. 
Might be worth starting to baseline to 22, as 23 will be out by the time we release AF5 fully anyway.

Reply:
Most certainly. We'll baseline AF5 towards the latest long term support JDK version once ready.